node/external/blockAPI/baseBlock.go

```diff
@@ -345,3 +354,164 @@ func bigIntToStr(value *big.Int) string {
 
 	return value.String()
 }
+
+func alteredAccountsMapToAPIResponse(alteredAccounts map[string]*outport.AlteredAccount, tokensFilter string, withMetadata bool) *common.AlteredAccountsForBlockAPIResponse {
+	response := &common.AlteredAccountsForBlockAPIResponse{
+		Accounts: make([]*common.AlteredAccountAPIResponse, 0),
+	}
+
+	for address, altAccount := range alteredAccounts {
+		apiAlteredAccount := &common.AlteredAccountAPIResponse{
+			Address: address,
+			Balance: altAccount.Balance,
+		}
+
+		if len(tokensFilter) > 0 {
+			attachTokensToAlteredAccount(apiAlteredAccount, altAccount, tokensFilter, withMetadata)
+		}
+
+		response.Accounts = append(response.Accounts, apiAlteredAccount)
+	}
+
+	return response
+}
+
+func attachTokensToAlteredAccount(apiAlteredAccount *common.AlteredAccountAPIResponse, altAccount *outport.AlteredAccount, tokensFilter string, withMetadata bool) {
+	for _, token := range altAccount.Tokens {
+		if !shouldAddTokenToResult(token.Identifier, tokensFilter) {
+			continue
+		}
+		if withMetadata {
+			apiAlteredAccount.Tokens = append(apiAlteredAccount.Tokens, token)
+			continue
+		}
+
+		apiAlteredAccount.Tokens = append(apiAlteredAccount.Tokens, &outport.AccountTokenData{
+			Identifier: token.Identifier,
+			Balance:    token.Balance,
+			Nonce:      token.Nonce,
+			Properties: token.Properties,
+			MetaData:   nil,
```

Comment:
Naming: `Metadata` vs. `MetaData`.

Reply:
I didn't change this struct on this PR

Reply:
in core we have:
`MetaData   *esdt.MetaData `json:"metadata"``
if change it, there will be an extra PR needed on elastic-indexer-go, so I'd rather not do it now